File: 3811-reverse-degree-of-a-string/reverse-degree-of-a-string.cpp
class Solution {
public:
    int reverseDegree(string s) {
        
        int sum = 0;

        for(int i=0; i < s.size(); i++){
            int idx = i + 1;
            int reverse_index = 'z' - s[i] + 1;
            sum += (idx * reverse_index);
        }
        return sum;
    }
};

File: 3811-reverse-degree-of-a-string/reverse-degree-of-a-string.java
class Solution {
    public int reverseDegree(String s) {
        
        int sum = 0;
        for(int i = 0; i<s.length(); i++){
            int pos = i + 1;
            int rev_pos = 'z' - s.charAt(i) + 1;

            sum += pos * rev_pos;
        }
        return sum;
    }
}